sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1055,60 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+
+class multinomial(CombinatorialFunction):
+    """
+    Implementation of multinomial function.
+    In mathematics, the multinomial theorem describes how to expand a power
+    of a sum in terms of powers of the terms in that sum. It is the
+    generalization of the binomial theorem from binomials to multinomials.
+
+    This is the number of ways of putting n1, n2, n3, n4....nm different objects
+    into m different boxes with nk in the kth box
+
+    Examples
+    ========
+
+    The third power of the trinomial a + b + c is given by
+
+    (a+b+c)^{3}=a^{3}+b^{3}+c^{3}+3a^{2}b+3a^{2}c ....
+    This can be computed by hand using the distributive property of
+    multiplication over addition, but it can also be done (perhaps more easily)
+    with the multinomial theorem, which gives us a simple formula for any
+    coefficient we might want. It is possible to "read off" the multinomial
+    coefficients from the terms by using the multinomial coefficient formula.
+
+    References
+    =========
+
+    [1] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/
+    [2] https://en.wikipedia.org/wiki/Multinomial_theorem
+    [3] https://dlmf.nist.gov/26.4
+    [4] https://www.maplesoft.com/support/help/Maple/view.aspx?path=combinat%2Fmultinomial
+
+    """
+
+    @classmethod
+    def eval(cls, *k):
+        k = [_sympify(x) for x in k]
+        n = Add(*k)
+        if all(x.is_Integer for x in k):
+            n = Add(*k)
+            numer = factorial(n)
+            denom = Mul(*(factorial(x) for x in k))
+            return numer/denom
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        n = Add(*args)
+        numer = factorial(n)
+        denom = Mul(*(factorial(x) for x in args))
+        return numer/denom
+
+    def _eval_rewrite_as_binomial(self, *k, **kwargs):
+        from sympy import binomial, Add, Mul
+        Mul(*[binomial(j,k) for j,k in zip([Add(*k[:i]) for i in range(1,len(k)+1)],k)])
```

Comment:
Move these imports at the top of the file and IMO there should be a `return` statement at line 1110

Reply:
that's because he plagiarized my code 🤣 ...